satpy/readers/generic_image.py

```diff
@@ -102,12 +106,27 @@ def end_time(self):
 
     def get_dataset(self, key, info):
         """Get a dataset from the file."""
-        logger.debug("Reading %s.", key)
-        return self.file_content[key['name']]
+        ds_name = self.dataset_name if self.dataset_name else key['name']
+        logger.debug("Reading '%s.'", ds_name)
+        data = self.file_content[ds_name]
+
+        # Mask data if necessary
+        try:
+            data = mask_image_data(data, info)
+        except ValueError as err:
+            logger.warning(err)
+
+        return data
 
 
-def mask_image_data(data):
-    """Mask image data if alpha channel is present."""
+def mask_image_data(data, info=None):
```

Comment:
When is `info` ever not provided?

Reply:
I don't know. This function is used in `test_generic_image.py`. That's why I fear it is part of the public interface and used by somebody somewhere..

Reply:
Whet Dave said. And indeed, this ought to be private.

Reply:
Done